tests/sflow/test_sflow.py

```diff
@@ -474,7 +474,7 @@ def testDelAgent(self, duthost, partial_ptf_runner):
         # Verify  whether the samples are received with previously configured agent ip
         partial_ptf_runner(
             polling_int=20,
-            agent_id=var['lo_ip'],
+            agent_id=var['mgmt_ip'],
```

Comment:
This is a change of behavior. Is this documented in the hsflowd release notes?

Reply:
As per this commit , [When dynamic agent or agentIP override is removed, revert to automati… · sflow/host-sflow@7cb2b83 (github.com)](https://github.com/sflow/host-sflow/commit/7cb2b83ef34ed970a7b6028e8b3d4b44ddef8452)  and as per the discussion https://github.com/sflow/host-sflow/pull/49#issuecomment-1410679089 , whenever the agent-id is removed, agent IP address chosen should be unique persistent and reachable and it can reach to the collector IP. Currently when we are executing this script we got this below error, 
![image](https://github.com/sonic-net/sonic-mgmt/assets/130334506/772e6905-b5ba-432f-b728-246921ec7eca)

 
After upgrading to 2.0.41, when the agent-id is deleted, it is expecting the management ip is the next preferred elected address to reach to the collector as per this script. So, we modified as per the behavior.  
Attached the failed logs for more information.
[test_sflow.log](https://github.com/sonic-net/sonic-mgmt/files/12511878/test_sflow.log)

Reply:
Looks like a hsflowd behavioral change. As long as it is documented, I am fine. @prsunny  @padmanarayana FYI